File: SharpBot.Service/MessageService.cs
using SharpBot.Data;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace SharpBot.Service
{
    public class MessageService
    {
        readonly MessageData _messageData;

        public MessageService(MessageData messageData)
        {
            _messageData = messageData;
        }

        public Task AddMessage(ulong guildId, ulong channelId, ulong authorId, string message, string[] attachments, DateTime dateTime)
        {
            string formatedContent = "";

            for (int i = 0; i < message.Length; i++)
            {
                if (message[i] == '\'')
                    formatedContent += "\\'";
                else
                    formatedContent += message[i];
            }

            if (attachments.Length > 0)
            {
                formatedContent += "\n";
                foreach (var attachment in attachments)
                    formatedContent += attachment + "\n";
            }

            return _messageData.AddMessage(
                guildId,
                channelId,
                authorId,
                formatedContent,
                dateTime);
        }
    }
}

File: Sharp.ServiceFSharp/MessageService.fs
namespace Sharp.FSharp.Service

open System

module MessageService =
    open Sharp.FSharp.Data

    let getMessages = MessageData.getMessages
    let addMessageWithAttachments guildId channelId userId date attachments message =  
        let formatedMessage =
            sprintf "%s\n\n%s"
                (attachments
                |> String.concat("\n"))
                message

        formatedMessage
        |> MessageData.addMessage guildId channelId userId date

    let addMessage guildId channelId userId date message =
        message
        |> MessageData.addMessage guildId channelId userId date
   